src/findings_data_import.py

```diff
@@ -49,12 +49,292 @@ def move_processed_file(s3_client, bucket, folder, filename):
+        with open(temp_data_filepath) as data_json_file:
+            findings_data = json.load(data_json_file)
+
+        logging.info("JSON data loaded from %s.", data_filename)
+        return temp_data_filepath, findings_data
+    except json.JSONDecodeError as json_err:
+        raise (
+            Exception(
+                f"Unable to decode JSON data for {data_filename}", json_err
+            ).with_traceback()
+        )
+    except ClientError as err:
+        raise (
+            Exception(
+                f"Error downloading file {data_filename} from {s3_bucket} ", err
+            ).with_traceback()
+        )
+
+
+def extract_findings(findings_data, field_map_dict):
+    """
+    Validate and return cleaned/processed finding.
+
+    Parameters
+    ----------
+    findings_data : dict
+        The findings dictionary pulled from a findings JSON file
+
+    field_map_dict: dict
+        The dictionary of replacement rules for field names in findings_data
+
+    Returns
+    -------
+    list : A list of findings objects from findings_data that pass validation
+
+    """
+    valid_findings = []
+
+    # if we (v2) get a lone object, wrap it in a list for compatability
+    if type(findings_data) != list:
+        findings_data = [findings_data]
+
+    # Iterate through data and save each record to the database
+    for index, finding in enumerate(findings_data):
+
+        if not finding or not hasattr(finding, "keys"):
+            logging.warning("Received an empty of invalid finding object, skipping.")
+            continue
+
+        # Replace or rename fields from replacement JSON
+        for field in field_map_dict:
+            if field in finding.keys():
+                if field_map_dict[field]:
+                    finding[field_map_dict[field]] = finding[field]
+                finding.pop(field, None)
+
+        # work with v1 and v2. If has NCATS ID OR findings the document is probably OK
+        if "RVA ID" not in finding.keys() or (
+            not ("NCATS ID" in finding.keys() and "Severity" in finding.keys())
+            and "findings" not in finding.keys()
+        ):
+            logging.warning(
+                "Skipping record %s. Missing 'RVA ID' or 'NCATS ID' field.", index
+            )
+            continue
+
+        # Get RVA ID in format DDDD([.-]D+) from the end of the "RVA ID" field.
+        rvaId = re.search(r"(\d{4})(?:[.-](\d+))?$", finding["RVA ID"])
+        if rvaId:
+            rID = f"RV{rvaId.group(1)}"
+            if rvaId.group(2) is not None:
+                rID += f".{rvaId.group(2)}"
+            finding["RVA ID"] = rID
+        else:
+            logging.warning(
+                "Skipping record %s: Unable to extract valid RVA ID from '%'",
+                index,
+                finding["RVA ID"],
+            )
+            continue
+
+        valid_findings.append(finding)
+
+    logging.info(
+        "{%s/%s} documents successfully processed.",
```

Comment:
Both variables are integers:
```suggestion
        "{%d/%d} documents successfully processed.",
```